### PR TITLE
Refactor Driver.drive and oc.compute.

### PR DESCRIPTION
**Pull request by lang-m**
- Splits the `Driver.drive` method into multiple smaller ones. Removes code duplication and helps in code structuring.

- `mif` files can now be written independently of running the actual simulation.

- A new method `Driver.schedule` can be used to submit the execution of OOMMF to a scheduling system. This allows users to interactively build up a system in a notebook and maybe use the `MinDriver` to find the desired initial state. Then, a subsequent long-running `TimeDrive` (e.g. adding some field or current and driving the system) can be scheduled using the clusters scheduling system. The input files are written from within the interactive notebook session and the OOMMF call is added to a user-provided schedule script (to define system requirements such as number of CPUs). The method does not modify the state of `system.m`. Therefore, it also allows users to submit multiple jobs starting from the same initial system, e.g. varying field/current strength.

  A call with slurm could look like `td.schedule(system, 'sbatch', 'custom-slurm-header.sh', t=10e-9, n=1000)`.

  The resulting data can be loaded and analysed in post using `micromagneticdata`.

### CODE REVIEW

File: oommfc/util.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+@contextlib.contextmanager
+def changedir(dirname):
+    """Context manager for changing directory."""
+    cwd = os.getcwd()
+    os.chdir(dirname)
+    try:
+        yield
+    finally:
+        os.chdir(cwd)
```
Comment (lang-m) [inline oommfc/util.py:13]: @marijanbeg Should we move this to `ubermagutil`?

File: oommfc/drivers/driver.py [hunk 2]
```diff
@@ -156,64 +150,266 @@ def drive(
...
+                system_dir.glob(f"{mode}*"), key=lambda p: int(p.name.split("-")[1])
+            )
+            next_number = int(last_existing_simulation.name.split("-")[1]) + 1
+        except ValueError:  # glob did not find any directories
+            next_number = 0
+        if next_number > 0 and not append:
+            raise FileExistsError(
+                f"Directory {system.name=} already exists. To "
+                "append drives to it, pass append=True."
+            )
```
Comment (lang-m) [inline oommfc/drivers/driver.py:390]: @marijanbeg The error message (was there already) says that the directory `system.name` (without the `/drive-XXX` or `/compute-XXX`) does already exist and we don't add to it when `append=False`. However, the old (and new) code only tests if there are already sub-directories of the type of simulation we are currently doing (only `drive-XXX` if we use `driver.drive` or only `compute-XXX` if we use `oc.compute`).

What is the desired behaviour here? Do we want to allow writing a compute (drive) run to a system directory that already has a drive (compute) run if `append=False`? Or do we want to make sure that the base directory `system.name` does not exist yet?

**Review by lang-m — COMMENTED** [review]
The docstrings are not fully updated yet.

**Review by samjrholt — APPROVED** [review]
I've had a quick go and all looks good to me!

**Pull request merged**